src/Logging/DataQueue.cpp

```diff
@@ -61,6 +61,14 @@ size_t DataQueue::getDataSize() {
 	return _writer->dataSize();
 }
 
+size_t DataQueue::getNumEventsInQueue() {
+	return _publishQueue->getNumEvents();
+}
+
+bool DataQueue::isCacheFull() {
+	return _publishQueue->getNumEvents() == _publishQueue->getFileQueueSize() + RAM_QUEUE_EVENT_COUNT;
```

Comment:
Any chance at all that getNumEvents() could be larger than the right side of this equation? Might be a good idea to use >=. Can't trust no one else's code, and I definitely don't trust my own code either.

Reply:
The documentation says that getNumEvents should return ram count + file count.  File count is set to 100 by default, but I've noticed that when the publish queue gets to be full, getNumEvents never exceeds 100, so I guess it's only returning the count of files in the file queue.

Either way it looks like the appropriate statement is just gonna be getNumEvents >= 100